### PR TITLE
Store authentication token in keychain. API fail without authorization token

### PR DESCRIPTION
**Pull request by mateuszmackowiak**
Created new KeychainStorage
Replacing authentication token storage in Keychain Storage
Added tests for CreateSessionAPIService
Added test for UpdateSessionParamsServiceTests

### CODE REVIEW

File: AirCasting/CoreData/Session.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+//  Session+CoreDataClass.swift
+//  AirCasting
+//
+//  Created by Lunar on 01/04/2021.
+//
+
+import Foundation
+import CoreData
+
+public typealias SessionUUID = UUID
```
[the remark this answers is not in the document]
Comment (mateuszmackowiak) [reply AirCasting/CoreData/Session.swift]: fixed and changed to a simple struct SessionUUID

File: AirCasting/CreateSessionViews/ChooseSessionTypeView.swift [hunk 2]
```diff
@@ -182,7 +180,7 @@ struct ChooseSessionTypeView: View {
     }
     
     private func createNewSession(isSessionFixed: Bool) {
-        sessionContext.sessionUUID = UUID().uuidString
+        sessionContext.sessionUUID = UUID()
```
[the remark this answers is not in the document]
Comment (mateuszmackowiak) [reply AirCasting/CreateSessionViews/ChooseSessionTypeView.swift]: the new SessionUUID default constructor will create a rawValue string from UUID().uuidString

File: AirCasting/Services/DownloadMeasurmentsService.swift [hunk 3]
```diff
@@ -7,32 +7,52 @@
...
                 try self?.update()
             } catch {
-                assertionFailure("Failed to call update at \(timer) \(error)")
+                assertionFailure("Failed to call update at \(tick) \(error)")
+            }
+        }
+    }
+
+    private func update() throws {
+        let request: NSFetchRequest<Session> = Session.fetchRequest()
```
[the remark this answers is not in the document]
Comment (mateuszmackowiak) [reply AirCasting/Services/DownloadMeasurmentsService.swift:41]: Added a predicate for only fixed sessions

File: AirCasting/CoreData/NSManagedObjectContext+Utils.swift [hunk 4]
```diff
@@ -26,12 +26,12 @@ extension NSManagedObjectContext {
         return new
     }
     
-    func newOrExisting<T: NSManagedObject>(uuid: String) -> T  {
+    func newOrExisting<T: NSManagedObject>(uuid: UUID) throws -> T  {
```
[the remark this answers is not in the document]
Comment (mateuszmackowiak) [reply AirCasting/CoreData/NSManagedObjectContext+Utils.swift]: changed to SessionUUID. Makes it less generic but I assume that other method `func newOrExisting<T: NSManagedObject & Identifiable>(id: T.ID) throws -> T  {` accepting id should cover most of the cases.

**Review by anna1901 — APPROVED** [review]
Looks great 🚀

File: AirCasting/ABConnector/AirBeam3Configurator.swift [hunk 5]
```diff
@@ -50,7 +50,7 @@ struct AirBeam3Configurator {
     
     // To configure fixed session we need to send authMessage first
     // We're generating unique String for session UUID and sending it with users auth token to the AB
-    private func configureFixed(uuid: String) {
+    private func configureFixed(uuid: SessionUUID) {
         guard let auth = userDefaults.string(forKey: "auth_token") else { return }
```
Comment (anna1901) [inline AirCasting/ABConnector/AirBeam3Configurator.swift:54]: should we use auth_token from defaults here or takie it from keychain?

**Pull request merged**